index.html

```diff
@@ -2622,178 +2649,285 @@ <h3><dfn>Writing or pushing content</dfn></h3>
+        <a data-cite="html">effective domain</a> of the {{Document/origin}} as
+        domain name in external type names when
+        <a href="#mapping-external-data-to-ndef">
+        creating external type records</a>. Therefore, applications MAY specify
+        only the colon and the external type name.
+      </p>
+      <p class="note">
+        The [[NFC-RTD]] requires that external type names are represented
+        with the URN prefix “`urn:nfc:ext:`”, e.g. when reading
+        <a>NDEF messages</a>. However, since <a>external type records</a> are
+        distinguished by having the <a>TNF FIELD</a> set to `0x04`, there is
+        no risk seen for type name clashing. Also, there are
+        <a href="https://www.w3.org/2001/tag/doc/URNsAndRegistries-50.html">
+        W3C TAG recommendations</a> to avoid using URNs in the Web.
+        Therefore, Web NFC does not use the URN prefix neither when reading
+        or writing <a>NDEF messages</a>, but requires the leading colon ('`:`')
+        before the type name.
+      </p>
+      <p>
+        To <dfn>validate external type</dfn> given a |input:USVString| run
+        these steps:
+      </p>
+      <ol class=algorithm id="validate-external-type">
+        <li>
+          If |input:string| is not a {{USVString}} or it is empty, or its
+          length exceeds 255 bytes, or it contains white spaces,
+          return `false`.
+        </li>
+        <li>
+          If |input| does not start with colon ('`:`'), return |input|.
+          <p class="issue">
+            Validating domain name needs an algorithm.
+          </p>
+        </li>
+        <li>
+          If |input| starts with colon ('`:`'), let |domain:string| be the
+          <a data-cite="url#concept-url-serializer">serialization</a> of the
+          <a data-cite="html">effective domain</a> of the {{Document/origin}}.
+        </li>
+        <li>
+          Return |domain| concatenated with |input|.
+        </li>
+      </ol>
+    </section>
 
-      <section><h3>Mapping empty record to NDEF</h3>
+    <section><h3>Validating local type</h3>
+      <p>
+        To <dfn>validate local type</dfn> given a |input:USVString| run
+        these steps:
+      </p>
+      <ol class=algorithm id="validate-local-type">
+        <li>
+          If |input| is not a {{USVString}} or its length exceeds 255 bytes,
+          return `false` and terminate these steps.
+        </li>
+        <li>
+          If |input| does not start with a lowercase letter or a number,
+          return `false`.
+        <li>
+          Return `true`.
+        </li>
+      </ol>
+    </section>
+
+    <section><h3>Mapping empty record to NDEF</h3>
       <div>
-        To <dfn>map empty record to NDEF</dfn> given a |record:NDEFRecordInit|,
-        run these steps:
+        To <dfn>map empty record to NDEF</dfn> given a |record:NDEFRecordInit|
+        and |ndef|, run these steps:
```

Comment:
Algorithm below should not state "Let ndefRecord be the notation for the NDEF record to be created by the UA." anymore if I'm reading this correctly.

Reply:
Hmm, it doesn't.

Reply:
But it did say "Let |ndef| be ...", right. Thanks for noticing.